// Copyright 2022-2024 ETSI SDG TeraFlowSDN (TFS) (https://tfs.etsi.org/)
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

//Example of connection
syntax = "proto3";
package connection;

import "google/protobuf/empty.proto";

service ConnectionServiceWithNotif {
  rpc CreateConnection (Connection) returns (google.protobuf.Empty) {}
  rpc ListConnection (google.protobuf.Empty) returns (ConnectionList) {}
  rpc GetBer(Connection) returns (stream Ber) {}
}

message Ber {
  double value = 1;
}

message Connection {
  string connectionId = 1;
  string sourceNode = 2;  
  string targetNode = 3;
  string sourcePort = 4;  
  string targetPort = 5;  
  uint32 bandwidth = 6;
  
  enum LayerProtocolName {
    ETH = 0;
    OPTICAL = 1;
  }

  LayerProtocolName layerProtocolName = 7;

}

message ConnectionList {
  repeated Connection connection = 1;
}
